### PR TITLE
Add unit test of TabulatedFluidProperties::generateTabulatedData

### LINKED ISSUE #11329: Test TabulatedFluidProperties data generation
## Rationale

The tabulated data generation capability of TabulatedFluidProperties is currently not tested at all, but it should be.

## Description

Only the reading capability of TabulatedFluidProperties is currently tested (either in the unit tests or the test suites of the modules). Although the data files that are used in these tests have been previously generated using TabulatedFluidProperties, the generation capability is not tested at all. 

Changes to `TabulatedFluidProperties::generateTabulatedData` could introduce mistakes that will not be caught by the automated testing. 

## Impact

More robust testing of TabulatedFluidProperties

### PR DESCRIPTION
**Pull request by cpgr**
Closes #11329

### CODE REVIEW

File: unit/src/TabulatedFluidPropertiesTest.C [hunk 1]
```diff
@@ -120,3 +122,25 @@ TEST_F(TabulatedFluidPropertiesTest, fromFile)
   REL_TEST("cv", _tab_fp->cv(p, T), _co2_fp->cv(p, T), 1.0e-4);
   REL_TEST("entropy", _tab_fp->s(p, T), _co2_fp->s(p, T), 1.0e-4);
 }
+
+// Test generation of tabulated fluid properties
+TEST_F(TabulatedFluidPropertiesTest, generateTabulatedData)
+{
+  // We always want to generate a new file in this test, delete existing file
+  std::remove("fluid_properties.csv");
```
Comment (andrsd) [inline unit/src/TabulatedFluidPropertiesTest.C]: You want to do this in `TearDown()`, see [advanced guide](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md) to the unit testing package.

File: unit/src/TabulatedFluidPropertiesTest.C [hunk 2]
```diff
@@ -120,3 +122,25 @@ TEST_F(TabulatedFluidPropertiesTest, fromFile)
...
+
+  Real p = 1.5e6;
+  Real T = 450.0;
+
+  // Generate the tabulated data
+  const_cast<TabulatedFluidProperties *>(_tab_gen_fp)->initialSetup();
```
Comment (andrsd) [inline unit/src/TabulatedFluidPropertiesTest.C]: Could this be moved into `SetUp()`? It is part of the setup rather then the test.

**Review by andrsd — CHANGES_REQUESTED** [review]

File: unit/src/TabulatedFluidPropertiesTest.C (hunk 1 above)
Comment (cpgr) [reply unit/src/TabulatedFluidPropertiesTest.C]: I've moved this to `TearDown()`

File: unit/src/TabulatedFluidPropertiesTest.C (hunk 2 above)
Comment (cpgr) [reply unit/src/TabulatedFluidPropertiesTest.C]: I tried this, but it seems that `SetUp()` is called before each test in this file, so this is setup multiple times.

There is a `SetUpTestCase()` that from reading the documentation is called only once, but I couldn't get it to work properly.

Comment (andrsd) [reply unit/src/TabulatedFluidPropertiesTest.C]: That sucks :-( Thanks for trying...

**Review by andrsd — APPROVED** [review]

**Pull request merged**